src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @brief ucc_team_create_from_parent is a nonblocking collective operation,
+ * which creates a new team from the parent team. If a participant intends to
+ * participate in the new team, it passes a TRUE value for the “included”
+ * parameter. Otherwise, it passes FALSE. The routine returns immediately after
+ * the post-operation. To learn the completion of the team create operation, the
+ * ucc_team_create_test operation is used. 
+
+ * @param [in]     my_ep         Endpoint of the process/thread calling the split operation
+ * @param [in]    parent_team    Parent team handle from which a new team handle is created
+ * @param [in]    included       Boolean variable indicating whether the
+ *                               process/thread participates in the newly created team
+ * @param [out]    new_ucc_team   Pointer to the new team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_create_from_parent(uint64_t my_ep, bool included,
+                                             ucc_team_h parent_team, 
+                                             ucc_team_h *new_ucc_team);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_get_size routine queries the size of the team. It reflects the
+ * number of unique endpoints in the team.
+ *
+ * @param [in]   ucc_team  Team handle
+ * @param [out]  size      The size of team as number of endpoints
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_size(ucc_team_h ucc_team, uint32_t *size);
+
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_get_my_ep routine queries and returns the endpoint of the
+ * participant invoking the interface.
+ *
+ * @param [out]  ep          Endpoint of the participant calling the routine
+ * @param [in]   ucc_team    Team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_my_ep(ucc_team_h ucc_team, uint64_t *ep);
+
+/**
+ * @ingroup UCC_TEAM
+ *
+ * @ref ucc_team_my_ep routine queries and returns all endpoints of all
+ * participants in the team.
+ *
+ * @param [out]     ep          List of endpoints
+ * @param [out]     num_eps     Number of endpoints
+ * @param [in]      ucc_team    Team handle
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_team_get_all_eps(ucc_team_h ucc_team, uint64_t **ep, uint64_t
+                                  *num_eps);
+
+
+/**
+ * *************************************************************
+ *                   Collectives Section 
+ * *************************************************************
+ */
+typedef enum {
+    UCC_DT_INT8 = 0,
+    UCC_DT_INT16,
+    UCC_DT_INT32,
+    UCC_DT_INT64,
+    UCC_DT_INT128,
+    UCC_DT_UINT8,
+    UCC_DT_UINT16,
+    UCC_DT_UINT32,
+    UCC_DT_UINT64,
+    UCC_DT_UINT128,
+    UCC_DT_FLOAT16,
+    UCC_DT_FLOAT32,
+    UCC_DT_FLOAT64,
+    UCC_DT_USERDEFINED,
+    UCC_DT_OPAQUE
```

Comment:
if OPAQUE (or user-defined) is chosen how do we pass the pointer? I seem to recall we considered making this a union of enum+pointer, but I'm not sure what was the conclusion. I'm asking because I don't see any other relevant field in `ucc_coll_buffer_info_t`.

Reply:
Aug 19th WG Discussion: 
Instead of enum+pointer, we will have a “user-defined” reduction filed to coll_op_args.